src/KeboolaComponent.php

```diff
@@ -0,0 +1,141 @@
+<?php declare(strict_types = 1);
+
+namespace Keboola\Component;
+
+use ErrorException;
+use Keboola\Component\Config\KeboolaConfig;
+use Keboola\Component\Config\KeboolaConfigDefinition;
+use Keboola\Component\Manifest\ManifestManager;
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use function error_reporting;
+use function file_get_contents;
+
+/**
+ * This is the core class that does all the heavy lifting. By default you don't need to setup anything. There are some
+ * extension points for you to use if you want to customise the behavior.
+ */
+class KeboolaComponent
```

Comment:
`Keboola\Component\KeboolaComponent`? To stejně u `Keboola\Component\Config\KeboolaConfig` (tam už chybí jen aby ta classa nebyla `KeboolaConfig` ale `KeboolaComponentConfig` :-)), nejde to udělat nějak hezčejc?

Reply:
Je to kvůli extendování. Typicky si uděláš v projektu `MyExtractor\Component` a abys nemusel tuhle třídu aliasovat, tak se jmenuje jinak. Čistě kvůli lepší Developer Experience.

Reply:
Jasně, chápu, proč ne třeba `BaseComponent` nebo `BaseConfig`?

Reply:
Jo, to je reasonable. Upraveno.